ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py

```diff
@@ -36,17 +37,33 @@ def ethernet_interfaces(self) -> dict | None:
                     # The l3_interfaces has already been filtered in _filtered_tenants
                     # to only contain entries with our hostname
                     for l3_interface in vrf["l3_interfaces"]:
+                        nodes_length = len(l3_interface["nodes"])
+                        if (
+                            len(l3_interface["interfaces"]) != nodes_length
+                            or len(l3_interface["ip_addresses"]) != nodes_length
+                            or ("descriptions" in l3_interface and "description" not in l3_interface and len(l3_interface["descriptions"]) != nodes_length)
+                        ):
+                            raise AristaAvdError(
```

Comment:
Should this error be used for input/data errors like this, or should it be used for internal AVD errors?  The message output by this error class states "An Error has occured in an arista.avd plugin" which sounds more like an internal error to me?

Reply:
This is any generic error not covered by the more specific ones (which inherit from this one) - here I am overwriting the default message so that's fine and will make sense to the user :)

Reply:
Cool, thanks for the explanation!